AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -210,8 +219,16 @@ public final class APIClient: APIClientProtocol {
     private func handle<R: Request>(
         _ result: URLSessionDownloadSuccess,
         _ request: R
-    ) -> HTTPResponse<R.ResponseType> where R.ResponseType == DownloadResponse {
+    ) throws -> HTTPResponse<R.ResponseType> where R.ResponseType == DownloadResponse {
         log(result: result, request: request)
+        
+        
+        try DispatchQueue.global(qos: .default).sync {
```

Comment:
`sync` will still block the main thread until the file is read :-)

Reply:
please don't use `sync`, as discussed offline, the global queue can still use ANY thread form the pool to execute this block of code including the parent thread which would lead to thread dead lock

Reply:
then i think ill just have to remove this DispatchQueue related stuff for the file handling, and just accept that 'sometimes' the file io stuff might happen on the main thread, at least we dont run the risk of deadlocking. if neither the nil delegate queue or this DispatchQueue.global can ensure that safety then im not sure what else we can do here

Reply:
> then i think ill just have to remove this DispatchQueue related stuff for the file handling, and just accept that 'sometimes' the file io stuff might happen on the main thread

sure, agreed